lib/Serialization/ModuleFileSharedCore.cpp

```diff
@@ -307,6 +308,30 @@ static ValidationInfo validateControlBlock(
       break;
     case control_block::SDK_NAME: {
       result.sdkName = blobData;
+
+      // Enable this check for tagged compiler or when the
+      // env var is set (for testing).
+      static const char* forceDebugPreSDKRestriction =
+        ::getenv("SWIFT_DEBUG_FORCE_SWIFTMODULE_PER_SDK");
+      if (version::Version::getCurrentCompilerVersion().empty() &&
```

Comment:
It required a little research for me to understand what this check is doing. I would find it helpful for this predicate to be wrapped in an API that returns a `bool` and documents what it means for a compiler to be tagged or not. I also see a couple other places where such a convenience could be used.

Reply:
Good idea, the link between the different version services was unclear too. I added the service in the latest commit.